src/formatDate.js

```diff
@@ -8,7 +8,85 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [aFrom, bFrom, cFrom, separator] = fromFormat;
+  const fromFormatArray = [aFrom, bFrom, cFrom];
+  const [aTo, bTo, cTo, newSeparator] = toFormat;
+  const toFormatArray = [aTo, bTo, cTo];
+  const dateinNumbers = date.split(separator);
+
+  const dateInfo = getDateFromOld(fromFormatArray, dateinNumbers);
+
+  return setDateOrderNew(toFormatArray, dateInfo).join(newSeparator);
```

Comment:
The `setDateOrderNew` function assumes that `dateInfo` contains all necessary date parts. Ensure that `getDateFromOld` correctly populates `dateInfo` with all required fields, or handle cases where some fields might be missing.